robusta_krr/core/abstract/strategies.py

```diff
@@ -43,7 +43,16 @@ def timeframe_timedelta(self) -> datetime.timedelta:
 _StrategySettings = TypeVar("_StrategySettings", bound=StrategySettings)
 
 ArrayNx2 = Annotated[NDArray[np.float64], Literal["N", 2]]
-ResourceHistoryData = dict[str, ArrayNx2]
+
+
+class ResourceHistoryData(pd.BaseModel):
+    query: str  # The query used to get the data
+    data: dict[str, ArrayNx2]  # Mapping: pod -> (time, value)
+
+    class Config:
+        arbitrary_types_allowed = True
```

Comment:
where is this used?

Reply:
This is because we have numpy type annotations inside pydantic, that is not supported. So we need to say to pydantic, that we can have some arbitraty types.